src/blue.ts

```diff
@@ -354,17 +359,17 @@ export function blueProxyFactory(env: MembraneBroker) {
     // future optimization: hoping that proxies with frozen handlers can be faster
     freeze(BlueDynamicProxyHandler.prototype);
 
-    function getBlueValue(red: RedValue): BlueValue {
+    function getBlueValue<T>(red: T): T {
         if (isNullOrUndefined(red)) {
             return red as BlueValue;
         }
         if (typeof red === 'function') {
-            return getBlueFunction(red);
+            return (getBlueFunction((red as unknown) as RedFunction) as unknown) as T;
```

Comment:
maybe @pmdartus knowns better... it is unfortunately that we have to cast to unknown then to the proper type two ways :(

Reply:
to my knowledge there's no escaping this.